### PR TITLE
Django Tutorial Part 8: User authentication and permissions

### PR DESCRIPTION
**Pull request by manhminno**

### CODE REVIEW

**Comment by manhminno** [comment]
ready

File: catalog/models.py [hunk 1]
```diff
@@ -82,11 +85,18 @@ class BookInstance(models.Model):
...
         ordering = ["due_back"]
+        permissions = (("can_mark_returned", "Set book as returned"),)
 
     def __str__(self):
         """String for representing the Model object."""
         return f"{self.id} ({self.book.title})"
 
+    @property
+    def is_overdue(self):
+        if self.due_back and date.today() > self.due_back:
```
Comment (huongnt-2545) [inline catalog/models.py]:
```suggestion
        return self.due_back and date.today() > self.due_back
```

File: catalog/templates/catalog/bookinstance_list_borrowed_user.html [hunk 2]
```diff
@@ -0,0 +1,20 @@
+{% extends "base_generic.html" %}
+{% load i18n %}
+
+{% block content %}
+    <h1>{% translate "Borrowed books" %}</h1>
+
+    {% if bookinstance_list %}
```
Comment (huongnt-2545) [inline catalog/templates/catalog/bookinstance_list_borrowed_user.html:6]: check indent
```suggestion
    {% if bookinstance_list %}
      <ul>
        {% for bookinst in bookinstance_list %}
          <li class="{% if bookinst.is_overdue %}text-danger{% endif %}">
            <a href="{% url 'book-detail' bookinst.book.pk %}">{{bookinst.book.title}}</a> ({{ bookinst.due_back }})
          </li>
        {% endfor %}
      </ul>
    {% else %}
      <p>{% translate "There are no books borrowed." %}</p>
    {% endif %}
```

File: catalog/views.py [hunk 3]
```diff
@@ -42,5 +43,20 @@ class BookDetailView(generic.DetailView):
...
+class LoanedBooksByUserListView(LoginRequiredMixin, generic.ListView):
+    """Generic class-based view listing books on loan to current user."""
+
+    model = BookInstance
+    template_name = "catalog/bookinstance_list_borrowed_user.html"
+    paginate_by = 10
+
+    def get_queryset(self):
+        return (
+            BookInstance.objects.filter(borrower=self.request.user)
```
Comment (huongnt-2545) [inline catalog/views.py]:
```suggestion
            BookInstance.objects.filter(borrower=self.request.user, status__exact="o").order_by("due_back")
```

**Comment by manhminno** [comment]
ready

File: catalog/templates/catalog/bookinstance_list_borrowed_user.html (hunk 2 above)
Comment (manhminno) [reply catalog/templates/catalog/bookinstance_list_borrowed_user.html:6]: vâng ạ, em chỉnh r ạ

**Review by huongnt-2545 — APPROVED** [review]
merge pull nha em

**Pull request merged**